components/hab/src/command/pkg/upload.rs

```diff
@@ -52,26 +52,27 @@ use retry::retry;
 /// * Fails if it cannot find a package
 /// * Fails if the package doesn't have a `.hart` file in the cache
 /// * Fails if it cannot upload the file
-pub fn start<P: AsRef<Path>>(
+pub fn start<T, U>(
     ui: &mut UI,
     url: &str,
     channel: Option<&str>,
     token: &str,
-    archive_path: &P,
-    key_path: &P,
-) -> Result<()> {
+    archive_path: T,
+    key_path: U,
+) -> Result<()>
+where
+    T: AsRef<Path>,
+    U: AsRef<Path>,
```

Comment:
What's the benefit of having two type aliases here instead of one, given that they're the the same?